### PR TITLE
chore : replacing AssertJ's deprecated asList() DSL method with asInstanceOf(InstanceOfAssertFactories.list(type.class))

### LINKED ISSUE #3289: ControllerViaPluginConfigurationEnricherTest : Replace AssertJ's deprecated `asList()` DSL method with `asInstanceOf(InstanceOfAssertFactories.list(type.class))`
### Component

None

### Task description

In the latest versions of Assertj `asList()` DSL method has been marked as deprecated:

https://github.com/eclipse-jkube/jkube/blob/e1cde0bd0d4ab1a659db7b1f7a884705048b27de/jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/ControllerViaPluginConfigurationEnricherTest.java#L172

users are advised to use `asInstanceOf(InstanceOfAssertFactories.list(type.class))` instead, we need to update `asList()` calls with this. 

### Expected Behavior
AssertJ's `asList()` is replaced with `asInstanceOf(InstanceOfAssertFactories.list(type.class))`

### Acceptance Criteria
- [ ] `asList()` DSL method call is replaced with  `asInstanceOf(InstanceOfAssertFactories.list(type.class))` in all places

### PR DESCRIPTION
**Pull request by aizwal9**
## Description

Fixes #3289

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [x] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] I have read the [contributing guidelines](https://www.eclipse.dev/jkube/contributing)
 - [x] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [x] My code follows the style guidelines of this project
 - [x] I Keep It Small and Simple: The smaller the PR is, the easier it is to review and have it merged
 - [x] I use [conventional commits](https://www.conventionalcommits.org/) in my commit messages
 - [x] I have performed a self-review of my code
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [x] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [ ] I have added tests that prove my fix is effective or that my feature works
 - [x] New and existing unit tests pass locally with my changes
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Comment by manusa** [comment]
### Eclipse JKube [CI Report](https://github.com/eclipse-jkube/ci)

Started new GH workflow run for https://github.com/eclipse-jkube/jkube/pull/3299 (_2024-08-02T19:40:55Z_)

:gear: [JKube E2E Tests (10221148086)](https://github.com/eclipse-jkube/ci/actions/runs/10221148086) :hourglass_flowing_sand:
- :hourglass_flowing_sand: [Bootstrap CI test run (#3299)](https://github.com/eclipse-jkube/ci/actions/runs/10221148086/job/28283007070)

File: jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/ControllerViaPluginConfigurationEnricherTest.java [hunk 1]
```diff
@@ -13,19 +13,15 @@
  */
 package org.eclipse.jkube.enricher.generic;
 
-import io.fabric8.kubernetes.api.model.Container;
-import io.fabric8.kubernetes.api.model.EnvVarBuilder;
-import io.fabric8.kubernetes.api.model.KubernetesList;
-import io.fabric8.kubernetes.api.model.KubernetesListBuilder;
-import io.fabric8.kubernetes.api.model.PodSpec;
-import io.fabric8.kubernetes.api.model.PodTemplateSpec;
+import io.fabric8.kubernetes.api.model.*;
```
Comment (rohanKanojia) [inline jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/ControllerViaPluginConfigurationEnricherTest.java:16]: Could you please revert these import changes?

File: jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/ControllerViaPluginConfigurationEnricherTest.java [hunk 2]
```diff
@@ -169,42 +165,42 @@ void enablesImageChangeTriggers() {
...
       .extracting("type")
       .containsExactlyInAnyOrder("ImageChange", "ConfigChange");
   }
 
   private void assertGeneratedListContainsDeploymentWithNameAndEnvVar(KubernetesListBuilder kubernetesListBuilder, String name) {
     assertThat(kubernetesListBuilder.build())
         .extracting(KubernetesList::getItems)
-        .asList()
+        .asInstanceOf(InstanceOfAssertFactories.list(StatefulSet.class))
         .singleElement(InstanceOfAssertFactories.type(Deployment.class))
```
Comment (rohanKanojia) [inline jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/ControllerViaPluginConfigurationEnricherTest.java:177]:
```suggestion
        .asInstanceOf(InstanceOfAssertFactories.list(Deployment.class)))
        .singleElement()
```

File: jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/ControllerViaPluginConfigurationEnricherTest.java [hunk 3]
```diff
@@ -169,42 +165,42 @@ void enablesImageChangeTriggers() {
...
+        .asInstanceOf(InstanceOfAssertFactories.list(EnvVar.class))
         .contains(new EnvVarBuilder().withName("FOO").withValue("bar").build());
   }
 
   private void assertGeneratedListContainsStatefulSetWithNameAndEnvVar(KubernetesListBuilder kubernetesListBuilder, String name) {
     assertThat(kubernetesListBuilder.build())
         .extracting(KubernetesList::getItems)
-        .asList()
+        .asInstanceOf(InstanceOfAssertFactories.list(StatefulSet.class))
         .singleElement(InstanceOfAssertFactories.type(StatefulSet.class))
```
Comment (rohanKanojia) [inline jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/ControllerViaPluginConfigurationEnricherTest.java:194]:
```suggestion
        .singleElement()
```

**Review by rohanKanojia — APPROVED** [review]

**Pull request merged**